### ISSUE TITLE
Render from the topmost dirty component downward

### ISSUE THREAD

**Issue by joshaber** [joshaber/Few.swift#38]
Currently each component re-renders itself when its state changes. This works, but it's less than ideal.

Consider a component A which renders a component B. If both update their state in the same runloop, component B will be rendered twice: once by itself because of its state update, and once when component A re-renders itself.

We should start each re-render from the topmost dirty component.

(I don't think there are any correctness problems here, but it's certainly less performant.)

**Comment by mrjjwright** [comment]
Yeah I noticed this.  Each Component enqueues a render when it's state changes and is adding a separate render run loop handler.   I noticed that in React there is a notion of component's owner versus it's parent.  Each component can have only one owner of course, the react component that gave it it's property.  There is a single special `ReactCurrentOwner.owner` that is set at the beginning of a render.   This points to the component where render was called and if render is called again when a render is in place and this property is already set a warning is thrown in dev mode.  At the end of the render it is set back to  null.  See: https://github.com/facebook/react/blob/master/src/core/ReactCompositeComponent.js#L784.  You probably already knew this but used this as an excuse to learn more.
